multi_ptr-cast/sycl-2.2/index.md

```diff
@@ -0,0 +1,242 @@
+multi_ptr<ElementTypeU, Space>
+  static_pointer_cast(const multi_ptr<ElementTypeT, Space>& multiPtr);
+
+// Performs a dynamic_cast of the contained pointer
+template <typename ElementTypeU, typename ElementTypeT, access::address_space Space>
+multi_ptr<ElementTypeU, Space>
+  dynamic_pointer_cast(const multi_ptr<ElementTypeT, Space>& multiPtr);
+
+// Performs a const_cast of the contained pointer
+template <typename ElementTypeU, typename ElementTypeT, access::address_space Space>
+multi_ptr<ElementTypeU, Space>
+  const_pointer_cast(const multi_ptr<ElementTypeT, Space>& multiPtr);
+
+// Performs a reinterpret_cast of the contained pointer
+template <typename ElementTypeU, typename ElementTypeT, access::address_space Space>
+multi_ptr<ElementTypeU, Space>
+  reinterpret_pointer_cast(const multi_ptr<ElementTypeT, Space>& multiPtr);
+
+} // namespace sycl
+} // namespace cl
+```
+
+In the table below, each function has the following template parameters:
+```cpp
+template <typename ElementTypeU, typename ElementTypeT, access::address_space Space>
+```
+
+| Member function | Description |
+|-----------------|-------------|
+| *`multi_ptr<ElementTypeU, Space> static_pointer_cast(const multi_ptr<ElementTypeT, Space>& multiPtr)`* | Performs a `static_cast` of the underlying pointer `ElementTypeT*` contained within `multiPtr` to `ElementTypeU*` and returns a new `multi_ptr` instance containing the cast pointer. The address space stays the same. This conversion is only valid if the `static_cast` from `ElementType*` to `ElementTypeU*` is valid. |
+| *`multi_ptr<ElementTypeU, Space> dynamic_pointer_cast(const multi_ptr<ElementTypeT, Space>& multiPtr)`* | Performs a `dynamic_cast` of the underlying pointer `ElementTypeT*` contained within `multiPtr` to `ElementTypeU*` and returns a new `multi_ptr` instance containing the cast pointer. The address space stays the same. This conversion is only valid if the `dynamic_cast` from `ElementType*` to `ElementTypeU*` is valid. |
+| *`multi_ptr<ElementTypeU, Space> const_pointer_cast(const multi_ptr<ElementTypeT, Space>& multiPtr)`* | Performs a `const_cast` of the underlying pointer `ElementTypeT*` contained within `multiPtr` to `ElementTypeU*` and returns a new `multi_ptr` instance containing the cast pointer. The address space stays the same. This conversion is only valid if the `const_cast` from `ElementType*` to `ElementTypeU*` is valid. |
+| *`multi_ptr<ElementTypeU, Space> reinterpret_pointer_cast(const multi_ptr<ElementTypeT, Space>& multiPtr)`* | Performs a `reinterpret_cast` of the underlying pointer `ElementTypeT*` contained within `multiPtr` to `ElementTypeU*` and returns a new `multi_ptr` instance containing the cast pointer. The address space stays the same. This conversion is only valid if the `reinterpret_cast` from `ElementType*` to `ElementTypeU*` is valid. |
+
+## Add `const` to existing functions taking a `multi_ptr`
+
```

Comment:
this is separate from the casting extension right? It looks like an errata to SYCL 1.2.1 to me

Reply:
I think you are correct, I'll update the PR to only focus on the vendor extensions